src/server/actions/Volunteers.ts

```diff
@@ -1,15 +1,33 @@
 import dbConnect from '@/utils/db-connect';
 import VolunteerSchema from '@/server/models/Volunteer';
-import { VolunteerResponse } from '@/types/dataModel/volunteer';
+import { VolunteerResponse, VolunteerEntity } from '@/types/dataModel/volunteer';
+
 import OrganizationSchema from '@/server/models/Organization';
 OrganizationSchema;
 
+/**
+ * Soft delete a volunteer
+ * @param volunteerId The Id of the volunteer to be deleted
+ * @returns The volunteer in the database before soft-deletion, or null
+ */
+export async function softDeleteVolunteer(
+  volunteerId: string
+): Promise<VolunteerEntity | null> {
+  await dbConnect();
+
+  const res: VolunteerEntity | null = await VolunteerSchema.findByIdAndUpdate(
```

Comment:
should this be `VolunteerResponse`?

Reply:
@andlrutt

Reply:
yes. thought I commented on this, may have forgotten to submit the review 😁